src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -520,6 +520,12 @@ private static FunctionMetadata ParseFunctionMetadata(string functionName, JObje
                 functionMetadata.IsExcluded = (bool)value;
             }
 
+            if (configMetadata.TryGetValue("autoPackageRestoreEnabled", StringComparison.OrdinalIgnoreCase, out value) &&
+                value.Type == JTokenType.Boolean)
+            {
+                functionMetadata.IsAutoPackageRestoreEnabled = (bool)value;
```

Comment:
Valid points. We can simplify this for now and, if needed, add a flag that would allow users to opt-out.